ash: Fix 4ba6c5d3 regression - missing nfile.expfname initializer in copynode()

diff -uBp a/shell/ash.c b/shell/ash.c
--- a/shell/ash.c
+++ b/shell/ash.c
@@ -8099,6 +8103,8 @@ calcsize(union node *n)
 	case NAPPEND:
 		calcsize(n->nfile.fname);
 		calcsize(n->nfile.next);
+		if (n->nfile.expfname)
+			funcstringsize += strlen(n->nfile.expfname) + 1;
 		break;
 	case NTOFD:
 	case NFROMFD:
@@ -8216,6 +8222,10 @@ copynode(union node *n)
 		new->nfile.fname = copynode(n->nfile.fname);
 		new->nfile.fd = n->nfile.fd;
 		new->nfile.next = copynode(n->nfile.next);
+		if (n->nfile.expfname)
+			new->nfile.expfname = nodeckstrdup(n->nfile.expfname);
+		else
+			new->nfile.expfname = NULL;
 		break;
 	case NTOFD:
 	case NFROMFD:
